### ISSUE TITLE
bug: problem with some propagations not concluding in the max steps.

### ISSUE THREAD

**Issue by asalzburger** [acts-project/acts#581]
PR #390 reveals a problem with some propagations not concluding in the max steps.

@XiaocongAi has seen this as well with fitting - I am not saying this one creates the problem, it probably reveals it better due to the new ErrorCode.

_Originally posted by @asalzburger in https://github.com/acts-project/acts/pull/390#issuecomment-724723322_

**Comment by paulgessinger** [comment]
Can you elaborate how this is related?

**Comment by asalzburger** [comment]
Yes, we have a looper:

![Unknown-2](https://user-images.githubusercontent.com/26623879/98687147-b202d380-2369-11eb-9338-1b690e504fed.png)

**Comment by asalzburger** [comment]
So, the loop protection does not kick in (will check why), and then the propagation runs out of attempts and the PropagationAlgorithm doesn't check for this at all (which is also bad).

**Comment by asalzburger** [comment]
Ha, looks like the loop protection is too low for this example .... 

```shell
15:38:22    PropagationA   ERROR     Propagation reached the step count limit of 1000 (did 1000 steps)
Momentum of the particle is : 0.302444
``` 

The particle is just above threshold is seems ...

**Comment by asalzburger** [comment]
Fixed by # #582.
